### DISCUSSION TITLE
DNS friendly names from pkg/namesgenerator/names-generator.go

### DISCUSSION THREAD

**Discussion by cmflynn** [moby/moby#48922] [General]
I'm sure this has been discussed somewhere, but I haven't been able to find it in issues/discussions.  regarding https://github.com/moby/moby/blob/master/pkg/namesgenerator/names-generator.go

when resolving container names with default configurations, container names with underscores are not dns resolvable. so you end up having to pass --name or --hostname. this can be a bit annoying when dealing with various docker sdks and their libs that try and make it easy for users to do container<>container networking using their container names. 

what was the rationale for this?

**Comment by thaJeztah** [comment]
I don't think container names were resolvable when the generated names where introduced, and containers could only be resolved through "linked" containers; the hyphen was used for making it easier to copy/paste the name; https://github.com/moby/moby/pull/2422#discussion_r7265387

**Reply by cmflynn** [reply]
any interest in changing this?
